src/LauncherManager.vala

```diff
@@ -139,6 +139,33 @@
         reposition_launchers ();
     }
 
+    private void update_launcher_entry (string sender_name, GLib.Variant parameters, bool is_retry = false) {
+        if (!is_retry) {
+            // Wait to let further update requests come in to catch the case where one application
+            // sends out multiple LauncherEntry-updates with different application-uris, e.g. Nautilus
+            Idle.add (() => {
+                update_launcher_entry (sender_name, parameters, true);
+                return false;
+            });
+
+            return;
+        }
+
+        string app_uri;
+        VariantIter prop_iter;
+        parameters.get ("(sa{sv})", out app_uri, out prop_iter);
+
+        var app_id = app_uri.replace ("application://", "");
+        app_to_launcher[app_id].perform_unity_update (prop_iter);
```

Comment:
A null check would be good here to avoid criticals in the case of something like granite-7-demo